library/manageiq_tag_assignment.py

```diff
@@ -0,0 +1,229 @@
+    choices: ['True', 'False']
+  ca_bundle_path:
+    description:
+      - the path to a CA_BUNDLE file or directory with certificates
+    required: false
+    default: null
+'''
+
+EXAMPLES = '''
+# Assign tag on a provider in ManageIQ
+  manageiq_tag_assignment:
+    tags:
+    - category: environment
+      name: prod
+    - category: owner
+      name: prod_ops
+    resource_name: 'OpenShift01'
+    resource: 'provider'
+    state: 'present'
+    miq_url: 'http://localhost:3000'
+    miq_username: 'admin'
+    miq_password: '******'
+'''
+
+
+class ManageIQTagAssignment(object):
+    """ ManageIQ object to execute tag assignments in manageiq
+
+    url            - manageiq environment url
+    user           - the username in manageiq
+    password       - the user password in manageiq
+    miq_verify_ssl - whether SSL certificates should be verified for HTTPS requests
+    ca_bundle_path - the path to a CA_BUNDLE file or directory with certificates
+    """
+
+    manageiq_entities = {
+        'provider': 'providers', 'host': 'hosts', 'vm': 'vms',
+        'category': 'categories', 'cluster': 'clusters', 'data store': 'data_stores',
+        'group': 'groups', 'resource pool': 'resource_pools', 'service': 'services',
+        'service template': 'service_templates', 'template': 'templates',
+        'tenant': 'tenants', 'user': 'users', 'blueprint': 'blueprints'
+    }
+    actions = {'present': 'assign', 'absent': 'unassign'}
+
+    def __init__(self, module, url, user, password, miq_verify_ssl, ca_bundle_path):
+        self.module   = module
+        self.api_url  = url + '/api'
+        self.user     = user
+        self.password = password
+        self.client   = MiqApi(self.api_url, (self.user, self.password), verify_ssl=miq_verify_ssl, ca_bundle_path=ca_bundle_path)
+        self.changed  = False
+
+    def find_entity_by_name(self, entity_type, entity_name):
+        """ Searches the entity name in ManageIQ.
+
+        Returns:
+            the entity id if it exists in manageiq, None otherwise.
+        """
+        entities_list = getattr(self.client.collections, entity_type)
+        return next((e.id for e in entities_list if e.name == entity_name), None)
+
+    def query_resource_tags(self, resource_type, resource_id):
+        """ Returns a set of the full tag names assigned to the resource
+        """
+        try:
+            url = '{api_url}/{resource_type}/{resource_id}/tags?expand=resources'.format(api_url=self.api_url, resource_type=resource_type, resource_id=resource_id)
+            response = self.client.get(url)
+        except Exception as e:
+            self.module.fail_json(msg="Failed to query {resource_type} tags: {error}".format(resource_type=resource_type, error=e))
+        tags = response.get('resources', [])
+        tags_set = set([tag['name'] for tag in tags])
+        return tags_set
+
+    def is_tag_assigned(self, resource_type, resource_id, category_name, tag_name, assigned_tags):
```

Comment:
You can drop resource_type, resource_id params now.
Perhaps even reduce this to a `full_tag_name(category_name, tag_name)` or `full_tag_name(tag_dict)` function, so caller would check `full_tag_name(...) in assigned_tags`?

LGTM otherwise.